src/go/MONIT/annotationManager.go

```diff
@@ -0,0 +1,931 @@
+	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
+		log.Fatalf("Error parsing the response body: %s", err)
+	}
+	return data
+
+}
+
+//
+// The following block of code was taken from (with few changes done)
+// https://github.com/dmwm/CMSMonitoring/blob/master/src/go/MONIT/monit.go#L657
+
+// helper function to add annotation
+func addAnnotation(data []byte) {
+	var headers [][]string
+	bearer := fmt.Sprintf("Bearer %s", configJSON.Token)
+	h := []string{"Authorization", bearer}
+	headers = append(headers, h)
+	h = []string{"Content-Type", "application/json"}
+	headers = append(headers, h)
+
+	rurl := fmt.Sprintf("%s%s", configJSON.GrafanaBaseURL, configJSON.CreateAnnotationAPI)
+	if configJSON.Verbose > 0 {
+		log.Println(rurl)
+	}
+	req, err := http.NewRequest("POST", rurl, bytes.NewBuffer(data))
+	if err != nil {
+		log.Fatalf("Unable to make request to %s, error: %s", rurl, err)
+	}
+	for _, v := range headers {
+		if len(v) == 2 {
+			req.Header.Add(v[0], v[1])
+		}
+	}
+	if configJSON.Verbose > 1 {
+		dump, err := httputil.DumpRequestOut(req, true)
+		if err == nil {
+			log.Println("request: ", string(dump))
+		}
+	}
+	timeout := time.Duration(configJSON.HTTPTimeout) * time.Second
+	client := &http.Client{Timeout: timeout}
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Fatalf("Unable to get response from %s, error: %s", rurl, err)
+	}
+	if configJSON.Verbose > 1 {
+		dump, err := httputil.DumpResponse(resp, true)
+		if err == nil {
+			log.Println("response:", string(dump))
+		}
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Printf("Unable to read JSON Data from Grafana Annotation POST API, error: %v\n", err)
+	}
+
+	if configJSON.Verbose > 1 {
+		log.Println("response Status:", resp.Status)
+		log.Println("response Headers:", resp.Header)
+		log.Println("response Body:", string(body))
+	}
+}
+
+//
+// The following block of code was taken from (with few changes done)
+// https://github.com/dmwm/CMSMonitoring/blob/master/src/go/MONIT/monit.go#L813
+func createAnnotation() {
+
+	if annotation != "" {
+		if tags != "" {
+			dashboards := findDashboard()
+			timeRanges := parseTimes(true) // True because we are creating annotation and time ranges should be in future
+			if trange == "" {
+				log.Fatalf("Unable to create Annotation. Please provide Time Range!\n")
+			}
+			if configJSON.Verbose > 0 {
+				log.Println("timeRanges", timeRanges)
+			}
+			for _, r := range dashboards {
+				rec := make(map[string]interface{})
+				rec["dashboardId"] = r["id"]
+				rec["time"] = timeRanges[0]
+				rec["timeEnd"] = timeRanges[1]
+				rec["tags"] = parseTags(tags)
+				rec["text"] = annotation
+				data, err := json.Marshal(rec)
+				if err != nil {
+					log.Fatalf("Unable to marshal the data %+v, error %v\n", rec, err)
+				}
+				if configJSON.Verbose > 0 {
+					log.Printf("Add annotation: %+v", rec)
+				}
+				addAnnotation(data)
```

Comment:
there is no check for error here

Reply:
it would be similar case as updateAnnotationHelper, there are different error checks inside the addAnnotation block.